pkg/setup/controller_context.go

```diff
@@ -91,8 +94,47 @@ func NewControllerContext(ctx context.Context, options *config.VirtualClusterCon
 		return nil, err
 	}
 
+	multiNsCacheConfig := getLocalCacheOptionsForMultiNsCache(options)
+	var localMultiNamespaceManager ctrl.Manager
+
+	// this means we need to start another manager
+	if len(multiNsCacheConfig.DefaultNamespaces) > 0 {
```

Comment:
Make separate caches / managers for each fromHost resource to avoid the necessary union of namespace watches which will lead to unnecessary watch requests.